plane/src/client/mod.rs

```diff
@@ -70,11 +69,17 @@ impl PlaneClient {
     pub fn drone_connection(
         &self,
         cluster: &ClusterName,
+        pool: Option<String>,
     ) -> TypedSocketConnector<MessageFromDrone> {
-        let addr = self
-            .controller_address
-            .join(&format!("/ctrl/c/{}/drone-socket", cluster))
-            .to_websocket_address();
+        let path = match pool {
+            Some(pool) => {
+                let encoded_pool: String =
+                    form_urlencoded::byte_serialize(pool.as_bytes()).collect();
+                format!("/ctrl/c/{}/p/{}/drone-socket", cluster, encoded_pool)
+            }
+            None => format!("/ctrl/c/{}/drone-socket", cluster),
+        };
+        let addr = self.controller_address.join(&path).to_websocket_address();
```

Comment:
If `pool` is `Some("")`, we want the second URL rather than the first.

I see a few options here:
1. call the default pool `"default"` instead of `""` so that we can use it in the path, and only construct URLs of the first form that includes the pool. (We should still accept both types of URLs for now, so that updates go smoothly when we have drones that are already running).
2. Manually handle the case of `Some("")` here
3. Stop using options for pool altogether and just use a string, where the empty string is equivalent to what `None` means now. Although it's nice to be able to distinguish between `Some/None` values, the problem here is that there are really two ways of representing the default "pool": Some("") and None.

I think I'm inclined towards 1, what do you think?

Reply:
Per offline discussion:

- Take path #3 
- Only generate URLs of the form `.../p/{}/drone-socket`, even if the drone pool is the empty string (default)
  - the resulting URL `../p//drone-socket` is a bit ugly but that doesn't matter, it's only used programmatically anyway
- Keep `/ctrl/c/{}/drone-socket` as a working endpoint for now, so that running drones/proxies do not break when the new controller is deployed.

Reply:
Per offline discussion (part 2): @paulgb suggests we go back to the _original query param approach_ to passing pool (rather than in the path as we have here). It's not as hard as previously thought to handle query params in reverse proxies. See also https://github.com/drifting-in-space/plane/pull/614#discussion_r1485369710